src/engine/renderer/glsl_source/skybox_fp.glsl

```diff
@@ -22,19 +22,56 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 /* skybox_fp.glsl */
 
+const float radiusWorld = 4096.0; // Value used by quake 3 skybox code
+
 uniform samplerCube	u_ColorMap;
-uniform vec3		u_ViewOrigin;
+uniform sampler2D	u_CloudMap;
+
+uniform bool        u_UseCloudMap;
+uniform float       u_CloudHeight;
+
+uniform mat4		u_TextureMatrix;
+
+uniform float		u_AlphaThreshold;
 
 IN(smooth) vec3		var_Position;
 
 DECLARE_OUTPUT(vec4)
 
+// Parametric cloud function used by quake 3 skybox code
+float ComputeCloudParametric( vec3 skyVec, float radiusWorld, float cloudHeight ) {
+	return (1.0 / 2.0 * dot( skyVec, skyVec ) ) * ( -2.0 * skyVec.z * radiusWorld
+			+ 2.0 * sqrt( skyVec.z * skyVec.z * radiusWorld * radiusWorld
+			+ 2.0 * skyVec.x * skyVec.x * radiusWorld * cloudHeight
+			+ skyVec.x * skyVec.x * cloudHeight * cloudHeight + 2.0 * skyVec.y * skyVec.y * radiusWorld * cloudHeight
+			+ skyVec.y * skyVec.y * cloudHeight * cloudHeight
+			+ 2.0 * skyVec.z * skyVec.z * radiusWorld * cloudHeight + skyVec.z * skyVec.z * cloudHeight * cloudHeight  ) );
+}
```

Comment:
Such large compute may exceeds the ALU limit of early R300 but anyway running the game on such hardware likely requires to use the `lowest` profile and then, disable the sky entirely. So I don't see this as a problem.

Reply:
Well, https://https://unvanquished.net/about/ says daemon has compatibility for OpenGL 2.1, while R300 doesn't seem to support versions above 2.0, so it's a miracle if it works at all hehe.